src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -522,38 +537,6 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
                 lengthForConstructor = elementCount;
             }
 
-            if (inPlaceTarget is null && !used)
-            {
-                // The caller has specified that we're creating a ReadOnlySpan expression that won't be used.
-                // We needn't emit anything.
-                return true;
-            }
-
-            if (elementCount == 0)
-            {
-                // The span is empty.  Optimize away the array.  This works regardless of the size of the type.
-                // (We could optimize this even for non-primitives, but it's not currently worthwhile.)
-
-                // If this is in-place initialization, call the default ctor.
-                if (inPlaceTarget is not null)
-                {
-                    EmitAddress(inPlaceTarget, Binder.AddressKind.Writeable);
-                    _builder.EmitOpCode(ILOpCode.Initobj);
-                    EmitSymbolToken(spanType, wrappedExpression.Syntax);
-                    if (used)
-                    {
-                        EmitExpression(inPlaceTarget, used: true);
-                    }
-                }
-                else
-                {
-                    // Otherwise, assign it to a default value / empty span.
-                    Debug.Assert(used);
-                    EmitDefaultValue(spanType, used, wrappedExpression.Syntax);
-                }
-                return true;
-            }
-
             if (IsPeVerifyCompatEnabled())
```

Comment:
> if (IsPeVerifyCompatEnabled()) [](http://example.com/codeflow?start=11&length=31)

Consider moving this check right after ```if (elementCount == 0)``` check. They were next to each other before